### ISSUE TITLE
Out of Memory Error when running 'geth account new' on Raspberry Pi

### ISSUE THREAD

**Issue by Famicoman** [ethereum/go-ethereum#1716]
Using geth-ARM-20150820194331-1.1.0-d51d002 binary on a Raspberri Pi Model B (Not an RPi 2) under Debian.

Logs below:

ethereum@raspberrypi ~ $ ./geth account new
Your new account is locked with a password. Please give a password. Do not forget this password.
Passphrase:
Repeat Passphrase:
fatal error: runtime: out of memory

runtime stack:
runtime.SysMap(0x11100000, 0x10000000, 0xb6d93101, 0xd90f58)
        /usr/lib/go/src/runtime/mem_linux.c:159 +0x134
runtime.MHeap_SysAlloc(0xd93d30, 0x10000000, 0x49d84)
        /usr/lib/go/src/runtime/malloc.c:284 +0x140
runtime.MHeap_Alloc(0xd93d30, 0x8000, 0x0, 0xb6d90101, 0x66bd4d)
        /usr/lib/go/src/runtime/mheap.c:240 +0x50

goroutine 1 [running]:
runtime.switchtoM()
        /usr/lib/go/src/runtime/asm_arm.s:193 +0x4 fp=0x11069388 sp=0x11069384
runtime.mallocgc(0x10000000, 0x695388, 0x1, 0x4000000)
        /usr/lib/go/src/runtime/malloc.go:194 +0x9f8 fp=0x110693e0 sp=0x11069388
runtime.newarray(0x695388, 0x4000000, 0x10fc3800)
        /usr/lib/go/src/runtime/malloc.go:365 +0xec fp=0x110693fc sp=0x110693e0
runtime.makeslice(0x6891f8, 0x4000000, 0x0, 0x4000000, 0x0, 0x0, 0x0, 0x0)
        /usr/lib/go/src/runtime/slice.go:32 +0x1fc fp=0x11069420 sp=0x110693fc
golang.org/x/crypto/scrypt.Key(0x11059880, 0xc, 0x10, 0x11014c20, 0x20, 0x20, 0x40000, 0x8, 0x1, 0x20, ...)
        /slave/build-go-ethereum-develop-arm/build/Godeps/_workspace/src/golang.org/x/crypto/scrypt/scrypt.go:235 +0x1b4 fp=0x110694ac sp=0x11069420
github.com/ethereum/go-ethereum/crypto.keyStorePassphrase.StoreKey(0x10f3f8c0, 0x21, 0x10f3f380, 0x10fef3e0, 0xc, 0x0, 0x0)
        /slave/build-go-ethereum-develop-arm/go/src/github.com/ethereum/go-ethereum/crypto/key_store_passphrase.go:90 +0xd4 fp=0x11069664 sp=0x110694ac
github.com/ethereum/go-ethereum/crypto.(*keyStorePassphrase).StoreKey(0x10f1d538, 0x10f3f380, 0x10fef3e0, 0xc, 0x0, 0x0)
        <autogenerated>:5 +0xdc fp=0x11069684 sp=0x11069664
github.com/ethereum/go-ethereum/crypto.GenerateNewKeyDefault(0xb6da19b8, 0x10f1d538, 0xb6da0e78, 0x10f0b060, 0x10fef3e0, 0xc, 0x10f3f380, 0x0, 0x0)
        /slave/build-go-ethereum-develop-arm/go/src/github.com/ethereum/go-ethereum/crypto/key_store_plain.go:61 +0xe4 fp=0x110696a8 sp=0x11069684
github.com/ethereum/go-ethereum/crypto.keyStorePassphrase.GenerateNewKey(0x10f3f8c0, 0x21, 0xb6da0e78, 0x10f0b060, 0x10fef3e0, 0xc, 0xb6da1970, 0x0, 0x0)
        /slave/build-go-ethereum-develop-arm/go/src/github.com/ethereum/go-ethereum/crypto/key_store_passphrase.go:64 +0xac fp=0x110696d8 sp=0x110696a8
github.com/ethereum/go-ethereum/crypto.(*keyStorePassphrase).GenerateNewKey(0x10f1d4e0, 0xb6da0e78, 0x10f0b060, 0x10fef3e0, 0xc, 0x18, 0x0, 0x0)
        <autogenerated>:1 +0xec fp=0x11069700 sp=0x110696d8
github.com/ethereum/go-ethereum/accounts.(*Manager).NewAccount(0x10f3f8f0, 0x10fef3e0, 0xc, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /slave/build-go-ethereum-develop-arm/go/src/github.com/ethereum/go-ethereum/accounts/account_manager.go:148 +0x8c fp=0x11069740 sp=0x11069700
main.accountCreate(0x10f79f80)
        /slave/build-go-ethereum-develop-arm/go/src/github.com/ethereum/go-ethereum/cmd/geth/main.go:632 +0xdc fp=0x110697d8 sp=0x11069740
github.com/codegangsta/cli.Command.Run(0x7cc698, 0x3, 0x0, 0x0, 0x0, 0x0, 0x0, 0x82f6d8, 0x14, 0xa3cb10, ...)
        /slave/build-go-ethereum-develop-arm/build/Godeps/_workspace/src/github.com/codegangsta/cli/command.go:97 +0xecc fp=0x11069a04 sp=0x110697d8
github.com/codegangsta/cli.(*App).RunAsSubcommand(0x10f0c880, 0x10f79d50, 0x0, 0x0)
        /slave/build-go-ethereum-develop-arm/build/Godeps/_workspace/src/github.com/codegangsta/cli/app.go:226 +0xb1c fp=0x11069b68 sp=0x11069a04
github.com/codegangsta/cli.Command.startApp(0x7d5748, 0x7, 0x0, 0x0, 0x0, 0x0, 0x0, 0x80bc10, 0xf, 0xa3eb70, ...)
        /slave/build-go-ethereum-develop-arm/build/Godeps/_workspace/src/github.com/codegangsta/cli/command.go:176 +0x388 fp=0x11069bb8 sp=0x11069b68
github.com/codegangsta/cli.Command.Run(0x7d5748, 0x7, 0x0, 0x0, 0x0, 0x0, 0x0, 0x80bc10, 0xf, 0xa3eb70, ...)
        /slave/build-go-ethereum-develop-arm/build/Godeps/_workspace/src/github.com/codegangsta/cli/command.go:97 +0x130c fp=0x11069de4 sp=0x11069bb8
github.com/codegangsta/cli.(*App).Run(0x10f0c800, 0x10f0a000, 0x3, 0x3, 0x0, 0x0)
        /slave/build-go-ethereum-develop-arm/build/Godeps/_workspace/src/github.com/codegangsta/cli/app.go:133 +0xd0c fp=0x11069f6c sp=0x11069de4
main.main()
        /slave/build-go-ethereum-develop-arm/go/src/github.com/ethereum/go-ethereum/cmd/geth/main.go:348 +0x94 fp=0x11069fc0 sp=0x11069f6c
runtime.main()
        /usr/lib/go/src/runtime/proc.go:63 +0x108 fp=0x11069fe4 sp=0x11069fc0
runtime.goexit()
        /usr/lib/go/src/runtime/asm_arm.s:1322 +0x4 fp=0x11069fe4 sp=0x11069fe4

goroutine 5 [syscall]:
os/signal.loop()
        /usr/lib/go/src/os/signal/signal_unix.go:21 +0x1c
created by os/signal.initÂ·1
        /usr/lib/go/src/os/signal/signal_unix.go:27 +0x40

goroutine 7 [chan receive]:
github.com/ethereum/go-ethereum/logger/glog.(*loggingT).flushDaemon(0xd86220)
        /slave/build-go-ethereum-develop-arm/go/src/github.com/ethereum/go-ethereum/logger/glog/glog.go:897 +0x78
created by github.com/ethereum/go-ethereum/logger/glog.initÂ·1
        /slave/build-go-ethereum-develop-arm/go/src/github.com/ethereum/go-ethereum/logger/glog/glog.go:428 +0x78

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
        /usr/lib/go/src/runtime/asm_arm.s:1322 +0x4

goroutine 8 [select]:
github.com/ethereum/go-ethereum/logger.dispatchLoop()
        /slave/build-go-ethereum-develop-arm/go/src/github.com/ethereum/go-ethereum/logger/sys.go:81 +0x57c
created by github.com/ethereum/go-ethereum/logger.initÂ·1
        /slave/build-go-ethereum-develop-arm/go/src/github.com/ethereum/go-ethereum/logger/sys.go:60 +0x34

**Comment by karalabe** [comment]
Hmm. It's been a long while since I played with running on an RPI v1, but its memory is very very limited (256 MB) and we haven't run many tests to try and fit into that. Any particular reason why you would want to run a full geth node on an RPI v1?

**Comment by Gustav-Simonsson** [comment]
@Famicoman This is a feature actually, it comes from using scrypt with parameters set rather high, it consumes an additional 256MB during account generation.

You can try adding a swap file manually to get around this:

``` bash
dd if=/dev/zero of=/PATH/FILENAME bs=1024 count=SIZE
mkswap /PATH/FILENAME
swapon /PATH/FILENAME
```

**Comment by Famicoman** [comment]
@karalabe I was an early adopter. I have a singe 256 MB model RPi, two 512 MB revisions, and a single Raspberry Pi 2. As these devices age, they slowly lose support so I start to wonder what I can do with them. I may just up to the 512 MB version for ethereum.

@Gustav-Simonsson I generated a 512 MB swap file and while it took a while, I got an account to generate. Thanks very much! I was hesitant of setting up a swap because of the toll it can take on an SD card, but that's nothing a swapoff can't fix after generation.
